nibabel/cifti2/parse_cifti2.py

```diff
@@ -0,0 +1,587 @@
+            self.struct_state.pop()
+            self.write_to = None
+
+        elif name == "BrainModel":
+            self.fsm_state.pop()
+            self.struct_state.pop()
+
+        elif name == "VertexIndices":
+            self.fsm_state.pop()
+            self.struct_state.pop()
+            self.write_to = None
+
+    def CharacterDataHandler(self, data):
+        """ Collect character data chunks pending collation
+
+        The parser breaks the data up into chunks of size depending on the
+        buffer_size of the parser.  A large bit of character data, with standard
+        parser buffer_size (such as 8K) can easily span many calls to this
+        function.  We thus collect the chunks and process them when we hit start
+        or end tags.
+        """
+        if self._char_blocks is None:
+            self._char_blocks = []
+        self._char_blocks.append(data)
+
+    def flush_chardata(self):
+        """ Collate and process collected character data
+        """
+        if self._char_blocks is None:
+            return
+        # Just join the strings to get the data.  Maybe there are some memory
+        # optimizations we could do by passing the list of strings to the
+        # read_data_block function.
+        data = ''.join(self._char_blocks)
+        # Reset the char collector
+        self._char_blocks = None
+        # Process data
+        if self.write_to == 'Name':
+            data = data.strip()  # .decode('utf-8')
+            pair = self.struct_state[-1]
+            pair[0] = data
+
+        elif self.write_to == 'Value':
+            data = data.strip()  # .decode('utf-8')
+            pair = self.struct_state[-1]
+            pair[1] = data
+
+        elif self.write_to == 'Vertices':
+            # conversion to numpy array
+            c = BytesIO(data.strip().encode('utf-8'))
+            vertices = self.struct_state[-1]
+            vertices.extend(np.genfromtxt(c, dtype=np.int))
+            c.close()
+
+        elif self.write_to == 'VoxelIndices':
+            # conversion to numpy array
+            c = BytesIO(data.strip().encode('utf-8'))
+            parent = self.struct_state[-1]
+            parent.voxel_indices_ijk.extend(np.genfromtxt(c, dtype=np.int))
+            c.close()
+
+        elif self.write_to == 'VertexIndices':
+            # conversion to numpy array
+            c = BytesIO(data.strip().encode('utf-8'))
+            index = self.struct_state[-1]
+            index.extend(np.genfromtxt(c, dtype=np.int))
+            c.close()
+
+        elif self.write_to == 'TransformMatrix':
+            # conversion to numpy array
+            c = BytesIO(data.strip().encode('utf-8'))
+            transform = self.struct_state[-1]
+            transform.matrix = np.genfromtxt(c, dtype=np.float)
+            c.close()
+
+        elif self.write_to == 'Label':
+            label = self.struct_state[-1]
+            label.label = data.strip().encode('utf-8')
```

Comment:
On python 3 this encoding means that the label attribute will be a bytes string instead of a unicode string. Before writing to XML this is converted into a str, leading to "b'<label name>'". Getting rid of the encode seems to work for me in both python 2 and 3